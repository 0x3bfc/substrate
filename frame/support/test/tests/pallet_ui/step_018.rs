#[frame_support::pallet]
mod pallet {
	use frame_support::pallet_prelude::*;

	#[pallet::trait_]
	pub trait Trait: frame_system::Trait {}

	#[pallet::module]
	pub struct Module<T> {}

	#[pallet::module_interface]
	impl<T: Trait> ModuleInterface for Module<T> {}

	#[pallet::call]
	impl Foo {}
}

fn main() {
}
