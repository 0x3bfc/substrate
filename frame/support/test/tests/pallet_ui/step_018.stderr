error: Invalid generics: expect `impl<T: Trait>` or `impl<T: Trait<I>, I: 'static>`
  --> $DIR/step_018.rs:15:2
   |
15 |     impl Foo {}
   |     ^^^^

error: unexpected end of input, expected `T`
 --> $DIR/step_018.rs:1:1
  |
1 | #[frame_support::pallet]
  | ^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
